mplc/utils.py

```diff
@@ -156,3 +167,88 @@ def set_log_file(path):
+    """
+    Create a graph of the partners' labels distribution per class
+
+    :param name: str, title of the graph and of the save file
+    :param ratio_partner_per_class: numpy array of shape (number of classes, number of partners),
+                                    ratio of the number of labels of each class per partner
+    :param corrupted_partners: dic (default is {} for no corruption), dictionary with the index of corrupted partners
+                                as key and the proportion of corruption as value
+    :param save_folder: pathlib Path (default is None, which implies no save of the figure), folder to save the figure
+    :param display: boolean (default is False), True to show the plot, False otherwise
+    """
+    if save_folder is None and not display:
+        warnings.warn(
+            "utils.create_data_distribution_graph() was called without setting the save_folder or the display option")
+
+    num_classes = len(ratio_partner_per_class)
+    num_partners = len(ratio_partner_per_class[0])
+
+    # Configure the plot
+    palette = sns.color_palette(palette="Set2", n_colors=num_partners)
+    hatches = {0.25: '.', 0.5: '..', 0.75: '...', 1: 'xx'}
+    hatch_color = 'darkred'
+    fig, ax = plt.subplots(figsize=(2.4 * num_classes, 1.33 * num_partners))
+
+    # Create a dataframe containing the partners' data distribution for each class and plot it
+    for i in range(num_classes - 1, -1, -1):
+        df = pd.DataFrame(data=ratio_partner_per_class[i, :], columns=['partner_part'])\
+               .assign(before_part=lambda df: df['partner_part'].cumsum().shift(fill_value=0))\
+               .assign(after_part=lambda df: 1 - df['partner_part'].cumsum())\
+               .reindex(['before_part', 'partner_part', 'after_part'], axis=1)\
+               .cumsum(axis=1).add(i, axis=1)\
+               .rename_axis('partner_num').reset_index()\
+               .melt(id_vars=['partner_num'], value_vars=['partner_part', 'before_part', 'after_part'],
+                     var_name='chunk', value_name='value')
+        for part, cols in zip(["after_part", "partner_part", "before_part"],
+                              [['white'] * num_partners, palette, ['white'] * num_partners]):
+            bars = sns.barplot(data=df.query('chunk=="' + part + '"'), y='partner_num', x='value',
+                               orient='h', palette=cols, ci=None, edgecolor=hatch_color, linewidth=0, )
+
+    # Add corruption hatches
+    corr_labels = [''] * num_partners
+    for corr, ratio in corrupted_partners.items():
+        selection = [c * num_partners * 3 + 1 * num_partners + corr for c in range(num_classes)]
+        keys = np.array(list(hatches.keys()))
+        hatch = hatches[keys[np.argmin(np.abs(keys - ratio))]]
+        for s in selection:
+            bars.patches[s].set_hatch(hatch)
+        corr_labels[corr] += f'\n({int(ratio * 100)}% corrupted data)'
```

Comment:
I am not experienced enough with Seaborn to understand this piece clearly - but it seems to me that dots representing corrupted data are applied to the whole bar of each class for which a partner has its data corrupted, right?